src/core/server/elasticsearch/client/cluster_client.ts

```diff
@@ -108,6 +109,7 @@ export class ClusterClient implements ICustomClusterClient {
     }
 
     return {
+      ...KIBANA_HEADERS,
```

Comment:
I wonder if we should also automatically forward this header from the incoming request. For instance, if elastic-agent makes a request to Kibana, what should the `x-elastic-product-origin` header be when it arrives at ES?

May be unnecessary for this initial change, but it could be useful for telemetry purposes in ES in the future.

Reply:
At the moment, this is only being consumed to determine when to suppress deprecation logs when the Kibana system-indices are queried by Kibana. If the usage of this grows, I could see us making this change, but it doesn't seem necessary at this point-in-time.